src/ucp/core/ucp_am.c

```diff
@@ -134,10 +134,10 @@ ucs_status_t ucp_worker_set_am_handler(ucp_worker_h worker, uint16_t id,
         goto out;
     }
 
-    worker->am.cbs[id].cb_old     = cb;
-    worker->am.cbs[id].context    = arg;
-    worker->am.cbs[id].flags      = flags;
-    worker->am.cbs[id].priv_flags = 0;
+    ucs_array_elem(&worker->am, id).cb_old     = cb;
+    ucs_array_elem(&worker->am, id).context    = arg;
+    ucs_array_elem(&worker->am, id).flags      = flags;
+    ucs_array_elem(&worker->am, id).priv_flags = 0;
```

Comment:
why need both flags and priv_flags?
maybe reuse same field but different bits for priv flags?